scipy/special/xsf/cdflib.h

```diff
@@ -0,0 +1,100 @@
+    if (std::isnan(p) || std::isnan(a) || std::isnan(x)) {
+        return std::numeric_limits<double>::quiet_NaN();
+    }
+    if (!((0 <= p) && (p <= 1))) {
+        set_error("gdtrib", SF_ERROR_DOMAIN, "Input parameter p is out of range");
+        return std::numeric_limits<double>::quiet_NaN();
+    }
+    if (!(a > 0) || std::isinf(a)) {
+        set_error("gdtrib", SF_ERROR_DOMAIN, "Input parameter a is out of range");
+        return std::numeric_limits<double>::quiet_NaN();
+    }
+    if (!(x >= 0) || std::isinf(x)) {
+        set_error("gdtrib", SF_ERROR_DOMAIN, "Input parameter x is out of range");
+        return std::numeric_limits<double>::quiet_NaN();
+    }
+    if (x == 0.0) {
+        if (p == 0.0) {
+            set_error("gdtrib", SF_ERROR_DOMAIN, "Indeterminate result for (x, p) == (0, 0).");
+            return std::numeric_limits<double>::quiet_NaN();
+        }
+        /* gdtrib(a, p, x) tends to 0 as x -> 0 when p > 0 */
+        return 0.0;
+    }
+    if (p == 0.0) {
+        /* gdtrib(a, p, x) tends to infinity as p -> 0 from the right when x > 0. */
+        set_error("gdtrib", SF_ERROR_SINGULAR, NULL);
+        return std::numeric_limits<double>::infinity();
+    }
+    if (p == 1.0) {
+        /* gdtrib(a, p, x) tends to 0 as p -> 1.0 from the left when x > 0. */
+        return 0.0;
+    }
+    double q = 1.0 - p;
+    auto func = [a, p, q, x](double b) {
+	if (p <= q) {
+	    return cephes::igam(b, a * x) - p;
+	}
+	return q - cephes::igamc(b, a * x);
+    };
+    double lower_bound = std::numeric_limits<double>::min();
+    double upper_bound = std::numeric_limits<double>::max();
+    /* To explain the magic constants used below:
+     * 1.0 is the initial guess for the root. -0.875 is the initial step size
+     * for the leading bracket endpoint if the bracket search will proceed to the
+     * left, likewise 7.0 is the initial step size when the bracket search will
+     * proceed to the right. 0.125 is the scale factor for a left moving bracket
+     * search and 8.0 the scale factor for a right moving bracket search. These
+     * constants are chosen so that:
+     *
+     * 1. The scale factor and bracket endpoints remain powers of 2, allowing for
+     *    exact arithmetic, preventing roundoff error from causing numerical catastrophe
+     *    which could lead to unexpected results.
+     * 2. The bracket sizes remain constant in a relative sense. Each candidate bracket
+     *    will contain roughly the same number of floating point values. This means that
+     *    the number of necessary function evaluations in the worst case scenario for
+     *    Chandrupatla's algorithm will remain constant.
+     *
+     * false specifies that the function is not decreasing. 342 is equal to
+     * max(ceil(log_8(DBL_MAX)), ceil(log_(1/8)(DBL_MIN))). An upper bound for the
+     * number of iterations needed in this bracket search to check all normalized
+     * floating point values.
+     */
```

Comment:
I'm putting this comment here for now, but rather than being duplicated in each function, I plan to explain the general principles with examples in the `xsf` documentation, which would be a deliverable for the SDG.